core/tests/puppeteer-acceptance-tests/puppeteer-testing-utilities/user-factory.js

```diff
@@ -77,7 +77,10 @@ let createNewBlogAdmin = async function(username) {
  * @returns The blog post editor instance created.
  */
 let createNewBlogPostEditor = async function(username) {
-  const blogAdmin = await createNewBlogAdmin('blogAdm');
+  let blogAdmin = activeUsers.find(user => user instanceof e2eBlogAdmin);
```

Comment:
@seanlip this is useful here. Say we already have a blogAdmin User in place, we can utilize that to create a blogPostEditor. However if activeUsers aren't been maintained we can't track who and how many users are currently in place. There can be cases where we are redefining a new blogAdmin every single time we need a new blogpostEditor. So i feel it's useful for now.

Reply:
No, don't do that. I know it will cause a performance degradation in the short term but we should fix this using composition rather than this hack. Services should be as stateless as possible.